src/mc-fle2-find-text-payload-private.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MC_FLE2_FIND_TEXT_PAYLOAD_PRIVATE_H
+#define MC_FLE2_FIND_TEXT_PAYLOAD_PRIVATE_H
+
+#include "mc-fle2-encryption-placeholder-private.h"
+#include "mongocrypt-buffer-private.h"
+#include "mongocrypt-private.h"
+#include "mongocrypt.h"
+
+#define DEF_TEXT_SEARCH_FIND_TOKEN_SET(Type)                                                                           \
```

Comment:
Since these are all the same, can't we just have one `mc_TextSearchFindTokenSet_t` type? If we /really/ want the split, we could just `typedef mc_TextSearchFindTokenSet_t mc_TextExactFindTokenSet_t`, etc. But that seems unnecessary.

Reply:
I want these to be distinct types, so that you won't be able to mindlessly assign one type to the other without the compiler yell at you.

Reply:
Sure... Even in that case, I think we should simplify some logic in the .c file.